File: jeong-hyuk/2월4일/교점에별만들기.py
def solution(line):
    dot_list = []
    
    # 직선들의 교점 찾기
    for i in range(len(line)):
        A, B, E = line[i]
        for j in range(i + 1, len(line)):
            C, D, F = line[j]
            
            bunja = A * D - B * C
            if int(bunja) == 0:
                continue

            x = (B * F - E * D) / bunja
            y = (E * C - A * F) / bunja

            # 정수인지 소수인지 확인
            if x.is_integer() and y.is_integer():
                x, y = int(x), int(y)
                dot_list.append([x, y])

    # 배열 크기 정하기
    x_dot_list = [dot[0] for dot in dot_list]
    y_dot_list = [dot[1] for dot in dot_list]
    x_min, x_max = min(x_dot_list), max(x_dot_list)
    y_min, y_max = min(y_dot_list), max(y_dot_list)

    # 배열 만들기
    arr = [['.' for _ in range(x_max - x_min + 1)] for _ in range(y_max - y_min + 1)]

    # 별 찍기
    for dot in dot_list:
        x, y = dot
        arr[y_max - y][x - x_min] = '*'

    return [''.join(a) for a in arr]

File: yht0827/2월/2월 4일/교점에 별만들기.java
import java.util.*;

class Solution {
	public List<String> solution(int[][] line) {
		List<String> answer = new ArrayList<>();
		int len = line.length;
		int Xmax = Integer.MIN_VALUE, Xmin = Integer.MAX_VALUE, Ymax = Integer.MIN_VALUE, Ymin = Integer.MAX_VALUE;
		List<List<Integer>> cor = new ArrayList<>();

		for (int i = 0; i < len; i++) {
			for (int j = i + 1; j < len; j++) {

				long a = line[i][0];
				long b = line[i][1];
				long e = line[i][2];

				long c = line[j][0];
				long d = line[j][1];
				long f = line[j][2];
				long denominator = (a * d) - (b * c);
				if (denominator == 0)
					continue;

				long numerator1 = (b * f) - (e * d);
				long numerator2 = (e * c) - (a * f);
				if (numerator1 % denominator != 0 || numerator2 % denominator != 0)
					continue;

				int x = (int)(numerator1 / denominator);
				int y = (int)(numerator2 / denominator);

				List<Integer> temp = Arrays.asList(x, y);

				if (!cor.contains(temp))
					cor.add(temp);

				Xmax = Math.max(Xmax, x);
				Xmin = Math.min(Xmin, x);
				Ymax = Math.max(Ymax, y);
				Ymin = Math.min(Ymin, y);
			}
		}

		for (int i = Ymin; i <= Ymax; i++) {
			StringBuilder sb = new StringBuilder();
			for (int j = Xmin; j <= Xmax; j++) {
				sb.append(".");
			}
			answer.add(sb.toString());
		}

		for (List<Integer> list : cor) {
			StringBuilder sb = new StringBuilder(answer.get(Math.abs(list.get(1) - Ymax)));
			sb.setCharAt(Math.abs(list.get(0) - Xmin), '*');
			answer.set(Math.abs(list.get(1) - Ymax), sb.toString());
		}

		return answer;
	}
}